src/core_modules/capture-core/components/WidgetProfile/DataEntry/factory/DataElementFactory.js

```diff
@@ -0,0 +1,196 @@
+// @flow
+/* eslint-disable no-underscore-dangle */
+import log from 'loglevel';
+import i18n from '@dhis2/d2-i18n';
+import { pipe, errorCreator } from 'capture-core-utils';
+
+import type {
+    CachedProgramTrackedEntityAttribute,
+    CachedTrackedEntityAttribute,
+    CachedOptionSet,
+} from '../../../../storageControllers/cache.types';
+import {
+    DataElement,
+    DateDataElement,
+    DataElementUnique,
+    dataElementUniqueScope,
+    dataElementTypes,
+} from '../../../../metaData';
+import { OptionSetFactory } from './OptionSetFactory';
+import { convertFormToClient, convertClientToServer } from '../../../../converters';
+import { getApi } from '../../../../d2/d2Instance';
+
+type ConstructorInput = {|
+    cachedTrackedEntityAttributes: Map<string, CachedTrackedEntityAttribute>,
+    cachedOptionSets: Map<string, CachedOptionSet>,
+|};
+
+export class DataElementFactory {
+    static errorMessages = {
+        TRACKED_ENTITY_ATTRIBUTE_NOT_FOUND:
+            'TrackedEntityAttributeId missing from programTrackedEntityAttribute or trackedEntityAttribute not found',
+    };
+
+    static buildtetFeatureType(featureType: 'POINT' | 'POLYGON') {
```

Comment:
is it `tei` instead of `tet`

Reply:
I think `tet` is correctly used here because the code is about the `tracked entity type`. Thanks!

Reply:
so can we make it camelCase? TET or Tet? Thanks